CRM/Gpapi/Processor.php

```diff
@@ -67,6 +67,18 @@ public static function preprocessContactData(&$params) {
       }
     }
 
+    if (!empty($params['gender_id']) && empty($params['prefix_id']) && empty($params['prefix'])) {
```

Comment:
The `Contact.create` API supports passing OptionValue names for parameters like `gender_id`, i.e. something like `cv api Contact.create gender_id="Male"`. This works for `Engage.signpetition` as well, however the prefix map would not work in that case.

We can make it work by running non-numeric values for `gender_id` through `CRM_Core_PseudoConstant::getKey()` first.